src/platform/packages/shared/kbn-unified-data-table/src/components/data_table.tsx

```diff
@@ -1137,6 +1150,35 @@ export const UnifiedDataTable = ({
     rowLineHeight: rowLineHeightOverride,
   });
 
+  const handleItemsRendered = useCallback(
+    ({ visibleRowStopIndex }: { visibleRowStopIndex: number }) => {
+      // visibleRowStopIndex is 0-based, rowCount is 1-based
+      if (visibleRowStopIndex === rowCount - 1) {
+        setHasScrolledToBottom(true);
+      } else {
+        setHasScrolledToBottom(false);
+      }
+    },
+    [rowCount]
+  );
+
+  const { run: throttledHandleItemsRendered } = useThrottleFn(handleItemsRendered, { wait: 500 });
+
+  const virtualizationOptions = useMemo(() => {
+    // Don't use row overscan when showing Document column since
+    // rendering so much DOM content in each cell impacts performance
+    if (defaultColumns) {
+      return {
+        onItemsRendered: throttledHandleItemsRendered,
+      };
+    }
+
+    return {
+      ...VIRTUALIZATION_OPTIONS,
+      onItemsRendered: throttledHandleItemsRendered,
```

Comment:
Q. Should this be undefined when `pagination === DEFAULT_PAGINATION_MODE`?
```suggestion
      onItemsRendered: pagination === DEFAULT_PAGINATION_MODE ?  undefined : throttledHandleItemsRendered,
```

Reply:
Given this function simply does a state (primitive) update and is already throttled, it did not make much performance difference if its execute for all modes. But now i am reading this code again, it would make sense to avoid that call. Goot catch 👍🏼